nixos/modules/programs/ghidra.nix

```diff
@@ -0,0 +1,45 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.programs.ghidra;
+in
+{
+  options.programs.ghidra = {
+    enable = lib.mkEnableOption "Ghidra, a software reverse engineering (SRE) suite of tools";
+
+    gdb = lib.mkOption {
+      default = false;
+      type = lib.types.bool;
+      description = ''
+        Whether to add to gdbinit the python modules required to make Ghidra's debugger work.
+      '';
+    };
+
+    package = lib.mkPackageOption pkgs "ghidra" { example = "ghidra-bin"; };
+  };
+
+  config = lib.mkIf (cfg.enable) {
+    environment = {
+      systemPackages = [ cfg.package ];
+
+      etc = lib.mkIf (cfg.gdb) {
+        "gdb/gdbinit.d/ghidra-modules.gdb".text = with pkgs.python3.pkgs; ''
+          python
+          import sys
+          [sys.path.append(p) for p in "${
+            (makePythonPath [
+              psutil
+              protobuf
+            ])
+          }".split(":")]
+          end
+        '';
```

Comment:
Would it make sense to create an option to allow adding more python modules to this list? Or would that be useless?

Reply:
Probably useless, ghidra only loads https://pypi.org/project/ghidragdb/ , so you only really need its dependencies to be available. 
We might want to add this as a general option to gdb though, but that's outside of the scope of this PR IMHO